### DISCUSSION TITLE
Setting up ejabberd behind nginx

### DISCUSSION THREAD

**Discussion by Eirikr70** [processone/ejabberd#4220] [Questions] [answered]
Hello, I have set up ejabberd as a docker container and it runs fine. My concern is that it is directly exposed to the Big Bad Web, not benefiting from the usual reverse-proxy. I have Swag as a reverse-proxy (nginx in a docker container), but it seems quite complex to connect. Standing behind the reverse-proxy would permit that ejabberd benefits from all my security setup, including crowdsec and suricata.
Has anyone set it up, who would be willing to share their settings ?
Thanks

**Comment by mremond** [comment, marked as answer]
Your proxy tool seems to be for HTTP right ?

If you are not using XMPP but either XMPP over Websockets or Bosh, your should be able to proxy it through nginx.

Otherwise you need to set a proxy at the TCP/IP level (Like for exemple HAProxy)

**Reply by Eirikr70** [reply]
The main point I would like to achieve is about the certificates. With Lestencrypt, I suppose I will have to renew my certs every 3 months right ? Then I suppose I have to forward router's port 80 to 5280-Ejabberd. But port 80 is routed to 80-Nginx. So for each renewal I would have to re-route port 80. I would like at least an easier and more automated way to renew my certs.
